### ISSUE TITLE
String interpolation

### ISSUE THREAD

**Issue by 11xx** [01mf02/jaq#75]
It would be great to print values with arbitraty text like jq's:
```
❯ echo '{"key1":"value1","key2":"value2"}' | jq '"The first key contains \(.key1), and the second, \"\(.key2)\""'
"The first key contains value1, and the second, \"value2\""

❯ echo '{"key1":"value1","key2":"value2"}' | jq -r '"The first key contains \(.key1), and the second, \"\(.key2)\""'
The first key contains value1, and the second, "value2"
```

_Source: [stedolan/jq#614](https://www.github.com/stedolan/jq/issues/614)_

**Comment by 01mf02** [comment]
I agree with you that this would be nice to have. The name of this functionality is called "string interpolation", and I have put it into the README among the jq functionality that jaq does not implement yet. However, I do not have the motivation to work on this myself. If you feel like implementing it, go ahead!

**Comment by nitsanavni** [comment]
Here's how I do it currently. Not as readable as I wished.

```jq
def interpolate: map(tostring) | add;
def s: if isarray then interpolate else tostring end;

def a: "a";
def b: "b";

# usage
["{ a: ", a, ", b: ", b, " }"] | s
```

**Comment by 01mf02** [comment]
@nitsanavni, well, in that case you could just do something like `"{ a: " + a + ", b: " + b + " }"`. :)

**Comment by pkoppstein** [comment]
I'd just like to point out that string interpolation isn't merely a convenience - it's essential for `sub` and `gsub` when used with named captures, e.g. 
```
'"a1bc2" | gsub("(?<x>[a-z])[0-9]"; "\(.x|ascii_upcase)")'
#=> "AbC"
```

**Comment by 01mf02** [comment]
@pkoppstein, couldn't you just write `'"a1bc2" | gsub("(?<x>[a-z])[0-9]"; .x|ascii_upcase)'`? Are there cases where this does not work?

**Comment by pkoppstein** [comment]
Yes.  Not having string interpolation is a source of such frustration that I wasn’t thinking clearly. My apologies.

**Comment by 01mf02** [comment]
> Yes. Not having string interpolation is a source of such frustration that I wasn’t thinking clearly. My apologies.

No worries. :)

**Comment by 01mf02** [comment]
This is now implemented in #109.
